stores/walmart.py

```diff
@@ -0,0 +1,390 @@
+                try:
+                    config = json.load(json_file)
+                    self.username = config["username"]
+                    self.password = config["password"]
+                    self.civ = config["civ"]
+                    self.url_groups = int(config["url_groups"])
+                    self.walmart_website = config.get("walmart_website", "walmart.com")
+                    for x in range(self.url_groups):
+                        self.url_list.append(config[f"url_list_{x+1}"])
+                        self.reserve.append(float(config[f"reserve_{x+1}"]))
+                    # assert isinstance(self.url_list, list)
+                except Exception:
+                    log.error(
+                        "walmart_config.json file not formatted properly: https://github.com/Hari-Nagarajan/nvidia-bot/wiki/Usage#json-configuration"
+                    )
+                    exit(0)
+        else:
+            log.error(
+                "No config file found, see here on how to fix this: https://github.com/Hari-Nagarajan/nvidia-bot/wiki/Usage#json-configuration"
+            )
+            exit(0)
+
+        for key in WALMART_URLS.keys():
+            WALMART_URLS[key] = WALMART_URLS[key].format(domain=self.walmart_website)
+        self.driver.get(WALMART_URLS["BASE_URL"])
+        log.info("Waiting for home page.")
+        if self.driver.title in CAPTCHA_URL:
+            self.defeat_captcha(self.driver.current_url)
+        if self.is_logged_in():
+            log.info("Already logged in")
+        else:
+            log.info("Lets log in.")
+            log.info("Wait for Sign In page")
+            self.login()
+            self.notification_handler.send_notification("Logged in and running", False)
+            log.info("Waiting 5 seconds.")
+            time.sleep(
+                5
+            )  # We can remove this once I get more info on the phone verification page.
+
+    def is_logged_in(self):
+        try:
+            time.sleep(5)
+            accountbutt = self.driver.find_element_by_xpath(
+                "/html/body/div[1]/div/div/div[1]/section/div[2]/div/div[3]/div[2]/div/div[2]/div[1]/button"
+            )
+            accountbutt.click()
+            signin = self.driver.find_element_by_xpath(
+                "/html/body/div[1]/div[1]/div/div[1]/section/div[3]/div[2]/div/div/div[1]/div/span"
+            ).text
+            log.info("Logged in account " + signin)
+            log.info("Checking login")
+            if signin == "Account":
+                return False
+            else:
+                return True
+        except Exception:
+            return False
+
+    def login(self):
+
+        signin = self.driver.find_element_by_xpath(
+            "/html/body/div[1]/div/div/div[1]/section/div[3]/div[2]/div/div/div[2]/div/a[1]/div/span/div"
+        )
+        signin.click()
+        time.sleep(1)
+
+        if self.driver.title == "Login":
```

Comment:
I am not sure wait_for_pages() will work. It is a frame that loads in from the side when you click the Account button. I added the wait because the menu fly-out wasn't happening before the click. Might be worth a bit of testing to shave some time off.

Reply:
yup, you are right - what you did makes sense here.